### PR TITLE
Make default error tolerance on positive relative to the value type

### LINKED ISSUE #11: `positive(val::T) -> T`?
Calling `positive(one(Float32))` should return `Positive{Float32}`, but it doesn't because `ε` is always a `Float64`. Maybe we should restrict the method to making `val` and `ε` the same type by default and make `ε=eps(T)` by default?

### PR DESCRIPTION
**Pull request by rofinn**
Closes #11

### CODE REVIEW

File: src/parameters.jl [hunk 1]
```diff
@@ -20,18 +20,19 @@ value(x::NamedTuple) = map(value, x)
 value(x::Dict) = Dict(k => value(v) for (k, v) in x)
 
 """
-    positive(val::Real, transform::Bijector=Bijectors.Exp(), ε::Real = 1e-12)
+    positive(val::T, transform::Bijector=Bijectors.Exp(), ε=sqrt(eps(T))) where T<:Real
```
Comment (rofinn) [inline src/parameters.jl:23]: Because we're changing the default tolerance this is technically a breaking change.

File: src/parameters.jl [hunk 2]
```diff
@@ -20,18 +20,19 @@ value(x::NamedTuple) = map(value, x)
...
 """
-function positive(val::Real, transform::Bijector=Bijectors.Exp(), ε::Real = 1e-12)
-    if val <= 0
-        throw(ArgumentError("Value, $val, is not positive."))
-    end
+function positive(
+    val::T, transform::Bijector=Bijectors.Exp(), ε=sqrt(eps(T)),
+) where T<:Real
+    val > 0 || throw(ArgumentError("Value ($val) is not positive."))
+    val > ε || throw(ArgumentError("Value ($val) is too small, relative to ε ($ε)."))
```
Comment (rofinn) [inline src/parameters.jl:35]: This could be one check/error, but we might want to distinguish them?

File: test/parameters.jl [hunk 3]
```diff
@@ -6,13 +6,17 @@ pdiagmat(args...) = PDiagMat(args...)
 @testset "parameters" begin
 
     @testset "postive" begin
-        @testset "$val" for val in [5.0, 1e-11, 1e-12]
+        @testset "$val" for val in [5.0, 0.001f0, 1.0e-7]
             p = positive(val)
             test_parameter_interface(p)
             @test value(p) ≈ val
+            @test typeof(value(p)) === typeof(val)
```
Comment (rofinn) [inline test/parameters.jl:13]: Test that value remains the same. This is the main thing we want to ensure here.

File: test/parameters.jl [hunk 4]
```diff
@@ -6,13 +6,17 @@ pdiagmat(args...) = PDiagMat(args...)
...
         end
 
+        # Test edge cases around the size of the value relative to the error tol.
         @test_throws ArgumentError positive(-0.1)
+        @test_throws ArgumentError positive(1e-11)
```
Comment (rofinn) [inline test/parameters.jl:18]: Since we have different error cases, lets make sure we hit both.

File: src/parameters.jl (hunk 2 above)
Comment (mattBrzezinski) [reply src/parameters.jl:35]: Personally I like how you wrote this, makes it very easy to read.

**Review by mattBrzezinski — APPROVED** [review]

File: src/parameters.jl (hunk 2 above)
Comment (willtebbutt) [reply src/parameters.jl:35]: I agree. This seems like the right way to do it.

**Review by willtebbutt — APPROVED** [review]
Yup, I'm also happy with this.

**Comment by rofinn** [comment]
Cool, thanks for reviewing so quickly :)
